src/visitors/IncorrectFunctionArgumentCountVisitor.ts

```diff
@@ -59,7 +60,7 @@ export class IncorrectFunctionArgumentCountVisitor extends Visitor {
 
             minimumArguments = functionMetadata.minimumArguments;
             // tslint:disable-next-line:max-line-length
-            assert(minimumArguments !== null && minimumArguments !== undefined, `TLE function metadata for '${actualFullFunctionName}' has a null or undefined minimum argument value.`);
+            assert(!isNullOrUndefined(minimumArguments), `TLE function metadata for '${actualFullFunctionName}' has a null or undefined minimum argument value.`);
```

Comment:
> assert(!isNullOrUndefined(minimumArguments), `TLE function metadata for '${actualFullFunctionName}' has a null or undefined minimum argument value.`); [](start = 11, length = 151)

So I'm kind of confused here. We removed "| undefined" from minimumArguments declaration. If functionMetadata.minimumArguments still has "| undefined" declared, why is this not causing a compile error? And do we need to check minimumArguments for both null and undefined?

Reply:
Good question.  I've declared the contract and the json schema for the metadata such that minimumArguments is number (not null). So theoretically it shouldn't be either.  But this does come from reading a JSON file, so I can't entirely guarantee it, esp. if we choose to read this file from the cloud later (we used to do that), so I'd rather still have the check in there.  I don't really like comparing against null and undefined directly, because usually we just check against both of them at the same time, using a falsy check - assert(minimumArguments). But here that's problematic because it could be the valid falsy value of zero. I just prefer the isNullOrUndefined way rather than two separate checks.